addons/project/models/project.py

```diff
@@ -805,7 +805,7 @@ def _compute_partner_id(self):
     @api.depends('partner_id.email', 'parent_id.email_from')
     def _compute_email_from(self):
         for task in self:
-            task.email_from = task.partner_id.email or task.email_from or task.parent_id.email_from
+            task.email_from = task.partner_id.email or task._origin.partner_id.email != task._origin.email_from and task.email_from or task.parent_id.email_from
```

Comment:
Unfortunately, your fix will not work once backported to 14.0. 

In 14.0 the field is not readonly, and can be changed by the user if a partner is set.
```suggestion
            task.email_from = task.partner_id.email or (task.partner_id and task.email_from) or task.parent_id.email_from
```